### PR TITLE
doc: add GOVERNANCE.md and clean up docs

### PR DESCRIPTION
**Pull request by AshCripps**
Add GOVERNANCE.md to the build working group

Fixes: #1319
Refs: https://github.com/nodejs/build/issues/2140#issuecomment-577620408

Ive used the example from the TSC doc: https://github.com/nodejs/TSC/blob/master/WORKING_GROUPS.md#bootstrap-governance

There is still some tweaks needed, feel free to comment/commit onto this branch.

cc: @nodejs/build @nodejs/tsc

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+or release process.
+
+The intention of the agenda is not to approve or review all patches;
+that should happen continuously on GitHub and be handled by the larger
+group of Collaborators.
+
+Any community member or contributor can ask that something be added to
+the next meeting's agenda by logging a GitHub Issue. Any Collaborator,
+WG member or the moderator can add the item to the agenda by adding
+the ***WG-agenda*** tag to the issue.
```
Comment (richardlau) [inline GOVERNANCE.md]:
```suggestion
the ***build-agenda*** tag to the issue.
```

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+### Build Working Group
+
+The Node.js Build Working Group is jointly governed by a Working Group (WG)
```
Comment (sam-github) [inline GOVERNANCE.md]: where was this copied from? "joint" would need to list multiple owners. This whole paragraph seems like its from some joint commcomm/node repo.

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+For the current list of Collaborators, see the project
+[README.md](./README.md#build-wg-members).
+
+### WG Membership
+
+WG seats are not time-limited.  There is no fixed size of the WG.
+
+There is no specific set of requirements or qualifications for WG
+membership beyond these rules. (I think this needs tweaking?)
```
Comment (sam-github) [inline GOVERNANCE.md]: this section needs replacing with https://github.com/nodejs/build/blob/master/doc/access.md#build-working-group-membership . but see my last comment, probably the whole file should be cloned from a more compatible repo

File: GOVERNANCE.md (hunk 2 above)
Comment (AshCripps) [reply GOVERNANCE.md]: In the description I link to where I got the original from, its from the TSC's bootstrap governance doc and is what the release doc is based on. I think jointly is in referral to each member of the group having a say in the governance of the working group.

File: GOVERNANCE.md (hunk 3 above)
Comment (AshCripps) [reply GOVERNANCE.md]: Will pull that document into this

File: GOVERNANCE.md (hunk 2 above)
Comment (sam-github) [reply GOVERNANCE.md]: I don't think lines 3-13 make sense for build wg.  https://github.com/nodejs/build#purpose is a better description

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+* Conduct guidelines
+* Maintaining the list of additional Collaborators
+
+For the current list of WG members, see the project
+[README.md](./README.md#current-project-team-members).
+
+### Collaborators
+
+The Build GitHub repository is
+maintained by the WG and additional Collaborators who are added by the
```
Comment (sam-github) [inline GOVERNANCE.md]: doesn't apply to build-wg

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+WG on an ongoing basis.
+
+Individuals making significant and valuable contributions are made
```
Comment (sam-github) [inline GOVERNANCE.md]: doesn't apply to build wg

File: GOVERNANCE.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+Collaborators and given commit-access to the project. These
+individuals are identified by the WG and their addition as
+Collaborators is discussed during the weekly WG meeting.
+
+_Note:_ If you make a significant contribution and are not considered
```
Comment (sam-github) [inline GOVERNANCE.md]: doesn't apply

File: GOVERNANCE.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+for commit-access log an issue or contact a WG member directly and it
+will be brought up in the next WG meeting.
+
+Modifications of the contents of the Build repository are made on
```
Comment (sam-github) [inline GOVERNANCE.md]: doesn't apply

File: GOVERNANCE.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+Collaborators. All pull requests must be reviewed and accepted by a
+Collaborator with sufficient expertise who is able to take full
+responsibility for the change. In the case of pull requests proposed
+by an existing Collaborator, an additional Collaborator is required
+for sign-off. Consensus should be sought if additional Collaborators
+participate and there is disagreement around a particular
+modification. See _Consensus Seeking Process_ below for further detail
+on the consensus model used for governance.
+
+Collaborators may opt to elevate significant or controversial
```
Comment (sam-github) [inline GOVERNANCE.md]: doesn't really apply, either, not wrong, but clearly is the nodejs consensus strategy

File: GOVERNANCE.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+There is no specific set of requirements or qualifications for WG
+membership beyond these rules. (I think this needs tweaking?)
+
+The WG may add additional members to the WG by unanimous consensus.
+
+A WG member may be removed from the WG by voluntary resignation, or by
+unanimous consensus of all other WG members.
+
+Changes to WG membership should be posted in the agenda, and may be
```
Comment (sam-github) [inline GOVERNANCE.md]: we don't do this, we PR it, and then accept it. nodejs does a formal announcement of new collabs in the TSC meetings

File: GOVERNANCE.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+suggested as any other agenda item (see "WG Meetings" below).
+
+If an addition or removal is proposed during a meeting, and the full
```
Comment (sam-github) [inline GOVERNANCE.md]: doesn't apply

**Comment by sam-github** [comment]
If we are going to have a governance, I think we'll have to do a bigger doc reorg, the current governance is organized across multiple files. doc/process/special access requests should be in here, too.

**Comment by rvagg** [comment]
yeah, this seems to double-up on existing docs, like https://github.com/nodejs/build/blob/master/doc/access.md
I'm fine if you want to take that on @AshCripps but I'd like to know that there's no duplication and that there's linkage between the various docs.

**Comment by AshCripps** [comment]
@rvagg its on my list to consolodate and remove duplicate files and also PR in and OFFBOARDING.md with a checklist and edit ONBOARDING to have the same checklist but in reverse.

**Comment by AshCripps** [comment]
Ive had a look through the repo and can't seem to spot any more duplication ive missed, Can anyone else spot anything missing from this PR?

**Comment by sam-github** [comment]
suggestions:
- README.md points to access.md (twice...), it should point to GOVERNANCE.md for joining
- access.md: "describes which groups have access to which assets managed by the  
Build Working Group and how membership of those groups is managed." ... GOVERNANCE describes the latter, I think.
- access.md TOC is broken, I think, "Build Working Group Membership" isn't in there anymore.
- I think access.md would be better renamed to "assets" or "resources", and the info about policy be moved to GOVERNANCE (such as the "We expect that even if you are... 1. 2. 3.", and "We consider infrat... vested interest in Node.js"), leaving just a list of resources with technical details about them.
- services.md... this seems to me like it should be merged with access.md, its not clear why its in a seperate file, so perhaps the remaining technical info in access.md should be in services.md?
- perhaps unrelated to this PR, but we have doc/non-ansible-configuration-notes.md and ansible/MANUAL_STEPS.md, both describe things that are not done by ansible, and both contain instructions for how to prep machines for ansible (doc/non-ansible-configuration-notes.md: "In order to get Windows machines to a state where Ansible can be run against them, some manual steps need to be taken"...). "nodejs.org / web host" is completely unrelated to the machines that ansible runs on, so it should perhaps stay where it is, but everything else in that file, AFIACAT, from "Windows (Azure/Rackspace)" on down is should be in ansible/MANUAL_STEPS.md, it describes thing that are done before or after ansible is run.
- doc/process/jenkins_job_configuration_access.md - I'm not sure what this is for, it appears to be partly a record of an ancient discussion. AFAICT, everything in it is either already in access.md, or should have been. After this change, it should mostly (all?) be in GOVERNANCE.md, and be deleted.

**Comment by AshCripps** [comment]
@nodejs/build Ive updated this PR and cleaned up a good chunk of docs, PTAL and review. 
Also ON/OFFBOARDING.md is resolved in #2142

File: doc/resources.md [hunk 11]
```diff
@@ -1,22 +1,24 @@
...
   * [ci.nodejs.org](#cinodejsorg)
     * [Jenkins admins](#jenkins-admins)
   * [ci-release.nodejs.org](#ci-releasenodejsorg)
   * [GitHub Bot](#github-bot)
+  * [email](#email)
 * [NPM Management](#npm-management)
 
-This document describes which groups have access to which assets managed by the
-Build Working Group and how membership of those groups is managed.
+This document describes resources and assets are managed by the
```
Comment (sam-github) [inline doc/resources.md]:
```suggestion
This document describes resources and assets managed by the
```

File: doc/resources.md [hunk 12]
```diff
@@ -103,6 +36,12 @@ GPG keys, so access to the repo does not itself give access to any of the
 secrets within. Individuals with different levels of access are able to use
 their GPG keys to decrypt a broader range of secrets.
 
+### Ansible configuration
+
+Ansible is the tool used to managed all machines that the Build WG is responsible for.
```
Comment (sam-github) [inline doc/resources.md]:
```suggestion
Ansible is the tool used to manage all machines that the Build WG is responsible for.
```

File: GOVERNANCE.md [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+The moderator is responsible for summarizing the discussion of each
+agenda item and sends it as a pull request after the meeting.
+
+### Communication
+
+The Build Working Group uses GitHub Issues to manage tasks. GitHub Pull Requests are used to suggest, and then land, changes.
+These Pull Requests must be:
+  * Keep open for at least 48 hours during the week (72 hours on the weekend)
+  * Get at least one approval from another WG member
+  * Land PRs with the same commit metadata you'd use in `nodejs/node`
```
[the remark this answers is not in the document]
Comment (AshCripps) [reply GOVERNANCE.md]: Changed

**Pull request merged**